10_rest/kasunfdo/cmd/puppy-server/main_test.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/anz-bank/go-course/10_rest/kasunfdo/pkg/puppy/store"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMain(m *testing.M) {
+	os.Args = []string{""}
+	os.Exit(m.Run())
+}
+
+func TestParseArgsLong(t *testing.T) {
+	args := []string{"--data", "../../data/data.json", "--port", "1234", "--store", "sync"}
+	config, err := parseArgs(args)
+	require.NoError(t, err)
+	assert.NotNil(t, config.dataFile)
+	assert.Equal(t, 1234, config.port)
+	assert.Equal(t, "sync", config.storeType)
+}
+
+func TestParseArgsShort(t *testing.T) {
+	args := []string{"-d", "../../data/data.json", "-p", "1234", "-s", "sync"}
+	config, err := parseArgs(args)
+	require.NoError(t, err)
+	assert.NotNil(t, config.dataFile)
+	assert.Equal(t, 1234, config.port)
+	assert.Equal(t, "sync", config.storeType)
+}
+
+func TestParseArgsWrong(t *testing.T) {
+	args := []string{"-f", "../../data/data.json", "-r", "1234", "-t", "sync"}
+	_, err := parseArgs(args)
+	assert.Error(t, err)
+}
+
+func TestParseArgsDefault(t *testing.T) {
+	var args []string
+	config, err := parseArgs(args)
+	require.NoError(t, err)
+	assert.NotNil(t, config.dataFile)
+	assert.Equal(t, 8080, config.port)
+	assert.Equal(t, "map", config.storeType)
+}
+
+func TestCreateStore(t *testing.T) {
+	storer, err := createStore("map")
+	require.NoError(t, err)
+	_, ok := storer.(*store.MapStore)
+	assert.True(t, ok)
+
+	storer, err = createStore("sync")
+	require.NoError(t, err)
+	_, ok = storer.(*store.SyncStore)
+	assert.True(t, ok)
+
+	storer, err = createStore("foo")
+	require.Error(t, err)
+	require.Nil(t, storer)
+	assert.Equal(t, "invalid input. given store type foo is invalid", err.Error())
+}
+
+func TestCreatePuppies(t *testing.T) {
+	s := store.NewMapStore()
+	dataFile, err := os.Open("../../data/data.json")
+	require.NoError(t, err)
+	err = createPuppies(dataFile, s)
+	assert.NoError(t, err)
+}
+
+type mockInvalidReader string
```

Comment:
`type mockInvalidReader struct {}`

I don't think you need to hold on to a string here, which would justify the string type aliasing, right?

Reply:
Yes! Thanks!!